velox/serializers/tests/PrestoSerializerTest.cpp

```diff
@@ -754,6 +754,51 @@ class PrestoSerializerTest
          makeMapVector<StringView, int32_t>({})});
   }
 
+  void testDeserializeFlatColumnIntoDictionaryVector(
+      const VectorPtr& alphabet,
+      const VectorPtr& flatVector) {
+    auto indices = makeIndices(alphabet->size() * 4, [&alphabet](auto row) {
+      return row % alphabet->size();
+    });
+    auto dictionary = BaseVector::wrapInDictionary(
+        nullptr, indices, alphabet->size() * 4, alphabet);
+
+    auto row = makeRowVector({flatVector});
+    std::ostringstream out;
+    serialize(row, &out, nullptr);
+    auto serialized = out.str();
+
+    auto byteStream = toByteStream(serialized);
+    auto result = makeRowVector({BaseVector::copy(*dictionary)});
```

Comment:
does `BaseVector::copy` create a dictionary vector? or is the copy flat?

Reply:
It copies flat, good call!